ush/python/pygfs/task/marine_analysis.py

```diff
@@ -0,0 +1,490 @@
+                'MARINE_WINDOW_MIDDLE_ISO': self.task_config.current_cycle.strftime('%Y-%m-%dT%H:%M:%SZ'),
+                'ENSPERT_RELPATH': _enspert_relpath,
+                'CALC_SCALE_EXEC': _calc_scale_exec,
+                'OPREFIX': f"{self.task_config.RUN}.t{self.task_config.cyc:02d}z."
+            }
+        )
+
+        # Extend task_config with local_dict
+        self.task_config.update(local_dict)
+
+    @logit(logger)
+    def initialize(self: Task) -> None:
+        """Initialize the marine analysis
+
+        This method will initialize the marine analysis.
+        This includes:
+        - staging the deterministic backgrounds (middle of window)
+        - staging SOCA fix files
+        - staging static ensemble members (optional)
+        - staging ensemble members (optional)
+        - generating the YAML files for the JEDI and GDASApp executables
+        - creating output directories
+        """
+        super().initialize()
+
+        # prepare the directory structure to run SOCA
+        self._prep_scratch_dir()
+
+        # fetch observations from COMROOT
+        # TODO(G.V. or A.E.): Keep a copy of the obs in the scratch fs after the obs prep job
+        self._fetch_observations()
+
+        # stage the ocean and ice backgrounds for FGAT
+        bkg_list = parse_j2yaml(self.task_config.MARINE_DET_STAGE_BKG_YAML_TMPL, self.task_config)
+        FileHandler(bkg_list).sync()
+
+        # stage the soca grid
+        FileHandler({'copy': [[os.path.join(self.task_config.COMIN_OCEAN_BMATRIX, 'soca_gridspec.nc'),
+                               os.path.join(self.task_config.DATA, 'soca_gridspec.nc')]]}).sync()
+
+        # link the flow dependent static B resources from the B-matrix task of the same cycle
+        os.symlink('../staticb', 'staticb')
+
+        # hybrid EnVAR case
+        if self.task_config.DOHYBVAR == "YES" or self.task_config.NMEM_ENS > 2:
+            # stage ensemble membersfiles for use in hybrid background error
+            logger.debug(f"Stage ensemble members for the hybrid background error")
+            mdau.stage_ens_mem(self.task_config)
+
+        # prepare the yaml configuration to run the SOCA variational application
+        self._prep_variational_yaml()
+
+        # prepare the yaml configuration to run the SOCA to MOM6 IAU increment
+        self._prep_checkpoint()
+
+    @logit(logger)
+    def _fetch_observations(self: Task) -> None:
+        """Fetch observations from COMIN_OBS
+
+        This method will fetch the observations for the cycle and check the
+        list against what is available for the cycle.
+        """
+
+        # get the list of observations
+        obs_list_config = YAMLFile(self.task_config.MARINE_OBS_LIST_YAML)
+        obs_list_config = Template.substitute_structure(obs_list_config, TemplateConstants.DOLLAR_PARENTHESES, self.task_config)
+        obs_list_config = {'observations': obs_list_config}
+        logger.info(f"{obs_list_config}")
```

Comment:
pretty sure we have stopped using `Template` and replaced with jinja.

Reply:
I think this will require making `gdas.cd/parm/soca/obs/obs_list.yaml` a jinja2 templated YAML and then making use of the `wxflow` `jinja` `replace_tmpl` filter.  You can follow, e.g.
https://github.com/NOAA-EMC/global-workflow/blob/ec634926f3b1ba1984b0735e17b76869659ceabe/parm/archive/master_enkf.yaml.j2#L32-L42
as an example.

Reply:
Actually, it is probably easier than that.  This can be done by changing
https://github.com/NOAA-EMC/GDASApp/blob/55e895f1dcf4e6be36eb0eb4c8a7995d429157e0/parm/soca/obs/obs_list.yaml#L1-L35
to the following jinja:
```jinja2
observers:
# ADT
- !INC {{MARINE_OBS_YAML_DIR}}/adt_rads_all.yaml

# SST
- !INC {{MARINE_OBS_YAML_DIR}}/sst_avhrr_ma_l3u.yaml
- !INC {{MARINE_OBS_YAML_DIR}}/sst_avhrr_mb_l3u.yaml
- !INC {{MARINE_OBS_YAML_DIR}}/sst_avhrr_mc_l3u.yaml
- !INC {{MARINE_OBS_YAML_DIR}}/sst_viirs_npp_l3u.yaml
- !INC {{MARINE_OBS_YAML_DIR}}/sst_viirs_n20_l3u.yaml
- !INC {{MARINE_OBS_YAML_DIR}}/sst_abi_g16_l3c.yaml
- !INC [{MARINE_OBS_YAML_DIR}}/sst_abi_g17_l3c.yaml
- !INC {{MARINE_OBS_YAML_DIR}}/sst_ahi_h08_l3c.yaml

# Ice concentration
- !INC {{MARINE_OBS_YAML_DIR}}/icec_amsr2_north.yaml
- !INC {{MARINE_OBS_YAML_DIR}}/icec_amsr2_south.yaml

# in situ: monthly
#- !INC {{MARINE_OBS_YAML_DIR}}/insitu_profile_bathy.yaml
- !INC {{MARINE_OBS_YAML_DIR}}/insitu_profile_argo.yaml
#- !INC {{MARINE_OBS_YAML_DIR}/insitu_profile_glider.yaml
#- !INC {{MARINE_OBS_YAML_DIR}}/insitu_profile_tesac.yaml
#- !INC {{MARINE_OBS_YAML_DIR}}/insitu_profile_tesac_salinity.yaml
#- !INC {{MARINE_OBS_YAML_DIR}}/insitu_profile_marinemammal.yaml
#- !INC {{MARINE_OBS_YAML_DIR}}/insitu_profile_xbtctd.yaml
#- !INC {{MARINE_OBS_YAML_DIR}}/insitu_surface_altkob.yaml
#- !INC {{MARINE_OBS_YAML_DIR}}/insitu_surface_trkob.yaml
#- !INC {{MARINE_OBS_YAML_DIR}}/insitu_surface_trkob_salinity.yaml

# in situ: daily
#- !INC {{MARINE_OBS_YAML_DIR}}/insitu_profile_dbuoy.yaml
#- !INC {{MARINE_OBS_YAML_DIR}}/insitu_profile_dbuoyb.yaml
#- !INC {{MARINE_OBS_YAML_DIR}}/insitu_profile_mbuoy.yaml
#- !INC {{MARINE_OBS_YAML_DIR}}/insitu_profile_mbuoyb.yaml
```

Reply:
We'll submit the changes for this in a subsequent PR, that obs_list.yaml cannot be jinja just yet. It's used elsewhere.

Reply:
https://github.com/NOAA-EMC/GDASApp/issues/1295